### PR TITLE
CI-CD Updates

### PR DESCRIPTION
**Pull request by Skptak**
Description
-----------
CI-CD Updates to use new versions of the CI-CD-Github-Actions
Individual PRs can be seen at:
[Complexity Check](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/70)
[Coverage-Cop](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/71)
[Executable Monitor](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/72)
[CSpell](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/75)
[Uncrustify Formatting Update](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/76)
[Link Verifier](https://github.com/FreeRTOS/CI-CD-Github-Actions/pull/81)

Test Steps
-----------

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [x] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -32,48 +32,48 @@ jobs:
...
       - name: Clone This Repo
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Run spellings check
-        uses: FreeRTOS/CI-CD-Github-Actions/spellings@main
+        uses: FreeRTOS/CI-CD-GitHub-Actions/spellings@v2
         with:
           path: ./
   formatting:
     runs-on: ubuntu-20.04
```
Comment (bradleysmith23) [inline .github/workflows/ci.yml:64]: Is there a reason this uses ubuntu 20.04 while others are using ubuntu-latest?

File: cspell.config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+---
+$schema: https://raw.githubusercontent.com/streetsidesoftware/cspell/main/cspell.schema.json
+version: '0.2'
+# Allows things like stringLength
+allowCompoundWords: true
```
Comment (bradleysmith23) [inline cspell.config.yaml:5]: are both uses of allow compound words necessary?

File: cspell.config.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+useGitignore: true
+languageSettings:
+  - allowCompoundWords: true
+    caseSensitive: false
+    enabled: true
+    languageId: c
+    locale: "*"
+# Could split this up? And do a dictionary for each repo?
```
Comment (bradleysmith23) [inline cspell.config.yaml:13]: Remove unnecessary comment.

**Review by bradleysmith23 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Skptak) [reply .github/workflows/ci.yml:64]: Uncrustify version 0.69.0_f that is used across all our repos is not the version installed on ubuntu-latest (ubuntu-22.04)

File: cspell.config.yaml (hunk 2 above)
Comment (Skptak) [reply cspell.config.yaml:5]: Fixed

File: cspell.config.yaml (hunk 3 above)
Comment (Skptak) [reply cspell.config.yaml]: Removed

**Review by archigup — APPROVED** [review]

**Review by bradleysmith23 — APPROVED** [review]

**Pull request merged**